src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferReadOnlySpanPropertiesOverReadOnlyArrayFieldsTests.cs

```diff
@@ -0,0 +1,503 @@
+        {
+            var test = new VerifyCS.Test
+            {
+                TestCode = $@"
+public class C
+{{
+    {declaration}
+}}",
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50
+            };
+            return test.RunAsync();
+        }
+
+        [Theory]
+        [InlineData("a = new byte[1];")]
+        [InlineData("(a, b) = t;")]
+        [InlineData("a = GetBytes();")]
+        [InlineData("a = new byte[] { 1, 2, 3 };")]
+        [InlineData("ConsumeBytesRef(ref a);")]
+        [InlineData("ref byte[] r = ref a;")]
+        public Task MutationInStaticCtor_NoDiagnostic_CS(string code)
+        {
+            var test = new VerifyCS.Test
+            {
+                TestCode = $@"
+using System;
+public class C
+{{
+    private static readonly byte[] a = new byte[] {{ 1 }};
+    private static byte[] GetBytes() => new byte[1];
+    private static void ConsumeBytesRef(ref byte[] bytes) {{ }}
+    private static byte b;
+    private static (byte[], byte) t;
+
+    static C()
+    {{
+        {code}
+    }}
+}}",
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50
+            };
+            return test.RunAsync();
+        }
+
+        [Theory]
+        [InlineData("byte[]")]
+        [InlineData("Span<byte>")]
+        public Task ReturnArrayOrSpan_NoDiagnostic_CS(string returnType)
+        {
+            string source = $@"
+using System;
+public class C
+{{
+    private static readonly byte[] a = new byte[] {{ 1, 2, 3 }};
+    private {returnType} M()
+    {{
+        return a;
+    }}
+}}";
+            var test = new VerifyCS.Test
+            {
+                TestCode = source,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50
+            };
+            return test.RunAsync();
+        }
+
+        [Fact]
+        public Task UsedInArrayInitializer_NoDiagnostic_CS()
+        {
+            var test = new VerifyCS.Test
+            {
+                TestCode = @"
+using System;
+public class C
+{
+    private static readonly byte[] a = new byte[] { 1, 2, 3 };
+    private static readonly byte[][] b = new byte[][]
+    {
+        a,
+        new byte[] { 4, 5, 6, 7 },
+        new byte[] { 8, 9 }
+    };
+}",
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50
+            };
+            return test.RunAsync();
+        }
+
+        [Fact]
+        public Task UsedInObjectInitializer_NoDiagnostic_CS()
+        {
+            var test = new VerifyCS.Test
+            {
+                TestCode = @"
+using System;
+public class O
+{
+    public byte[] A { get; set; }
+    public int I { get; set; }
+}
+
+public class C
+{
+    private static readonly byte[] a = new byte[] { 1, 2, 3 };
+    public void M()
+    {
+        var o = new O { A = a, I = 12 };
+    }
+}",
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50
+            };
+            return test.RunAsync();
+        }
+
```

Comment:
The overall test coverage looks great, only suggestions: please add a test with a comment